content/departments/cloud/technical-docs/v1.1/mi1-1_onprem_data_migration.md

```diff
@@ -0,0 +1,239 @@
+Note that these commands are intended for use as reference - you may need to adjust the commands for your deployment.
+
+USAGE
+        src [-v] snapshot databases <pg_dump|docker|kubectl> [--targets=<docker|k8s|"targets.yaml">]
+
+TARGETS FILES
+        Predefined targets are available based on default Sourcegraph configurations ('docker', 'k8s').
+        Custom targets configuration can be provided in YAML format with '--targets=target.yaml', e.g.
+
+                primary:
+                        target: ...   # the DSN of the database deployment, e.g. in docker, the name of the database container
+                        dbname: ...   # name of database
+                        username: ... # username for database access
+                        password: ... # password for database access - only include password if it is non-sensitive
+                codeintel:
+                        # same as above
+                codeinsights:
+                        # same as above
+
+        See the pgdump.Targets type for more details.
+```
+
+Each of the generated commands must be run to completion to generate a database dump for each database.
+The output is as follows:
+
+- `src-snapshot/primary.sql`
+- `src-snapshot/codeintel.sql`
+- `src-snapshot/codeinsights.sql`
+
+For custom or complex database setups, the operator will decide how best to proceed, in collaboration with IE/CSE/etc - the goal in the end is to generate the above database dumps in a format aligned with the output of `src snapshot databases pg_dump` (the plain `pg_dump` commands).
+
+> NOTE: Database export may affect the performance of the Sourcegraph instance while it is in progress. Depending on the size of the instance's databases, this can take a very long time.
+
+#### Instance summary
+
+A snapshot summary is used to run acceptance tests post-migration. The customer should create one with `src snapshot summary` - note that a site admin access token is required:
+
+```sh
+src login # configure credentials for the instance
+src snapshot summary
+```
+
+This will generate a JSON file at `src-snapshot/summary.json`. See `src snapshot summary --help` for more details.
+
+### Create migration resources
+
+First, the operator must [create an instance](mi1-1_creation_process.md) with the configuration for the desired final Cloud instance and freeze it:
+
+```sh
+mi ssh-exec 'cd /deployment/docker-compose && docker-compose down'
+```
+
+In the [`cloud-data-migrations`](https://github.com/sourcegraph/cloud-data-migrations) repository:
+
+1. Copy the `template/` directory, naming it corresponding to the customer
+2. For `project/`:
+   1. Fill out all `$CUSTOMER` variables and set all unset variables in `terraform.tfvars` as documented
+   2. Commit and push your changes
+   3. Create Terraform Cloud workspace for the directory and apply
+3. Then do the same for `resources/`, using the outputs of `project/`
```

Comment:
```suggestion
In the [`cloud-data-migrations`](https://github.com/sourcegraph/cloud-data-migrations) repository:

> The TFC workspace requires special configuration to properly set it up, we will document this in the future.

1. Copy the `template/` directory, naming it corresponding to the customer
2. For `project/`:
   1. Fill out all `$CUSTOMER` variables and set all unset variables in `terraform.tfvars` as documented
   2. Commit and push your changes
   3. Create Terraform Cloud workspace for the directory and apply
3. Then do the same for `resources/`, using the outputs of `project/`
```

Reply:
we need to manually configure the VCS stuff

Reply:
Updated suggestion and applied!